src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingProviderConfig.java

```diff
@@ -29,8 +29,46 @@ interface EmbeddingProviderConfig {
     @JsonProperty
     String apiKey();
 
+    /**
+     * A map of supported authentications. HEADER, SHARED_SECRET and NONE are the only techniques
+     * the DataAPI supports (i.e. the key of map can only be HEADER, SHARED_SECRET or NONE).
+     *
+     * @return
+     */
     @JsonProperty
-    List<String> supportedAuthentication();
+    Map<String, AuthenticationConfig> supportedAuthentication();
```

Comment:
nit: should maybe be "supportedAuthentications" (plural) since it can contain multiple entries, not just one